libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserFailureTests.java

```diff
@@ -40,37 +40,27 @@ public void testEntitlementDoesNotExist() {
     public void testEntitlementMissingParameter() {
         PolicyParserException ppe = expectThrows(PolicyParserException.class, () -> new PolicyParser(new ByteArrayInputStream("""
             entitlement-module-name:
-              - file: {}
+              - files:
```

Comment:
Optional: can we also add a test for an empty "file data"? (empty list)? 
Looking at the implementation, seems like an empty list would work (and it's the same as no entitlement), but I'm not sure if this is useful or not. A test however would "document" this one way or another (it's supported and it's like no entitlement, or it gives an error)

Reply:
I've updated this so the build method the parser uses doesn't allow for an empty path. We can just use constructor directly for tests that need to bypass this check.